src/main/scanLogic/scanManager.ts

```diff
@@ -46,4 +51,43 @@ export class ScanManager implements ExtensionComponent {
         let scanLogic: GraphScanLogic = new GraphScanLogic(this._connectionManager);
         return scanLogic.scan(graphRoot, flatten, progress, checkCanceled);
     }
+
+    /**
+     * Scan Cve in files for applicability issues.
+     * @param directory - the directory that will be scan
+     * @param abortController - the abort controller for cancele request
+     * @param cveToRun - the cve list we want to run applicability scan on
+     * @param skipFolders - the folders inside directory we want to skip scanning
+     * @returns the applicability scan response
+     */
+    public async scanApplicability(
+        directory: string,
+        abortController: AbortController,
+        cveToRun: string[] = [],
+        skipFolders: string[] = []
+    ): Promise<ApplicabilityScanResponse | undefined> {
+        let applicableRunner: ApplicabilityRunner = new ApplicabilityRunner(ScanManager.BINARY_ABORT_CHECK_INVTERVAL, this._logManager);
+        if (!applicableRunner.isSupported) {
+            return undefined;
```

Comment:
We should avoid returning undefined as much as possible. Let's log what is the problem as well

Reply:
done